src/Templates/ArchiveProductTemplatesCompatibility.php

```diff
@@ -182,60 +175,60 @@ function( $hook ) {
 	protected function set_hook_data() {
 		$this->hook_data = array(
 			'woocommerce_before_main_content'         => array(
-				'block_name' => 'core/query',
-				'position'   => 'before',
-				'hooked'     => array(
+				'block_names' => array( 'core/query', 'woocommerce/product-collection' ),
```

Comment:
This is my personal preference, but I would keep the same data structure across the compatibility layers. So, I would update the SingleProductTemplateCompatibility file as well.

Reply:
I agree, addressed in [this commit](https://github.com/woocommerce/woocommerce-blocks/pull/10332/commits/bfadbd47ffefd2afed69718e6a7298fa60721b23).

Reply:
Could you update [this comment](https://github.com/woocommerce/woocommerce-blocks/blob/6339f97e70b207a677f2f568c9b749c790eccc4c/src/Templates/AbstractTemplateCompatibility.php/#L99-L121) in the AbstractTemplateCompatibility?